### PR TITLE
feat(dao) new dao using kong.db, new schema, ttl

### PR DESCRIPTION
**Pull request by darrenjennings**
* schema updates
* kong storage adapter to new use new dao
* fix ml cache reference
* add migrations with ttl column

TODO:
- [x] fix travis to point to kong 1.0

### CODE REVIEW

File: spec/02-kong_storage_adapter_spec.lua [hunk 1]
```diff
@@ -217,8 +236,10 @@ for _, strategy in helpers.each_strategy() do
 
         assert.response(res).has.status(200)
 
+        local found = db.sessions:select_by_session_id(sid)
```
Comment (kinman) [inline spec/02-kong_storage_adapter_spec.lua]: Is the fact that the status code is 200 good enough, or should we also assert that there was no error executing this query? `found` can be nil for two reasons: 1) it's not in the db, and 2) the select query bombed out.

File: spec/02-kong_storage_adapter_spec.lua [hunk 2]
```diff
@@ -48,44 +59,52 @@ for _, strategy in helpers.each_strategy() do
...
+        plugins = "bundled, session",
         database   = strategy,
         nginx_conf = "spec/fixtures/custom_nginx.template",
       })
     end)
 
-    teardown(function()
-      helpers.stop_kong()
+    lazy_teardown(function()
+      helpers.stop_kong(nil, true)
```
Comment (kinman) [inline spec/02-kong_storage_adapter_spec.lua]: Why preserving servroot here?

File: spec/02-kong_storage_adapter_spec.lua [hunk 3]
```diff
@@ -8,12 +8,19 @@ function get_sid_from_cookie(cookie)
 end
 
 
-for _, strategy in helpers.each_strategy() do
+for _, strategy in helpers.each_strategy({'postgres'}) do
```
Comment (kinman) [inline spec/02-kong_storage_adapter_spec.lua]: Testing postgres only?

File: spec/01-access_spec.lua [hunk 4]
```diff
@@ -2,12 +2,18 @@ local utils = require "kong.tools.utils"
 local helpers = require "spec.helpers"
 
 
-for _, strategy in helpers.each_strategy() do
+for _, strategy in helpers.each_strategy({'postgres'}) do
```
Comment (kinman) [inline spec/01-access_spec.lua]: Postgres only?

File: kong/plugins/session/migrations/000_base_session.lua [hunk 5]
```diff
@@ -0,0 +1,37 @@
+return {
+  postgres = {
+    up = [[
+      CREATE TABLE IF NOT EXISTS sessions(
+        id            uuid,
+        session_id    text UNIQUE,
+        expires       int,
+        data          text,
+        created_at    timestamp without time zone default (CURRENT_TIMESTAMP(0) at time zone 'utc'),
+        ttl           timestamp with time zone,
```
Comment (kinman) [inline kong/plugins/session/migrations/000_base_session.lua]: It is counter-intuitive to me that created_at and ttl aren't the type of timestamp. I assume we have this pattern elsewhere as well?

Comment (darrenjennings) [reply kong/plugins/session/migrations/000_base_session.lua]: yea good catch looks like we switched all created_at to `with time zone` so just made them the same.

**Pull request merged**